ocaml/xapi/nm.ml

```diff
@@ -468,12 +468,25 @@ let bring_pif_up ~__context ?(management_interface=false) (pif: API.ref_PIF) =
       end;
 
       (* sync MTU *)
+      begin
       try
         let mtu = Int64.of_int (Net.Interface.get_mtu dbg ~name:bridge) in
         if mtu <> rc.API.pIF_MTU then
           Db.PIF.set_MTU ~__context ~self:pif ~value:mtu
       with _ ->
         warn "could not update MTU field on PIF %s" rc.API.pIF_uuid
+      end;
+      
+      (* sync igmp_snooping_enabled *)
+      if (rc.API.pIF_VLAN = -1L) && (rc.API.pIF_bond_slave_of = Ref.null) && (rc.API.pIF_tunnel_access_PIF_of == []) then begin
```

Comment:
Why not replace it with
match (get_pif_type) rc with
| `bond_pif _
| `phy_pif when rc,API.pIF_bond_slave_of = Ref.null -> balabalabala
| _ -> ()

Reply:
I agree with @krizex to use the helper function. We don't need to test for bond slaves, since this function is never called for bond slaves. Also, we should not exclude tunnel access PIFs, because tunnels have separate bridges, so the IGMP snooping status is relevant for those.

Reply:
For the case that this is a VLAN PIF, we should ensure here that the VLAN slave's is IGMP field is up-to-date.